Sources/SourceControl/GitRepository.swift

```diff
@@ -31,7 +46,7 @@ private struct GitShellHelper {
     /// Private function to invoke the Git tool with its default environment and given set of arguments.  The specified
     /// failure message is used only in case of error.  This function waits for the invocation to finish and returns the
     /// output as a string.
-    func run(_ args: [String], environment: EnvironmentVariables = Git.environment, outputRedirection: TSCBasic.Process.OutputRedirection = .collect) throws -> String {
+    func run(_ args: [String], environment: EnvironmentVariables, outputRedirection: TSCBasic.Process.OutputRedirection) throws -> String {
```

Comment:
have we lost the default env variables?

Reply:
There is a little change in the design, which makes the diff misleading indeed. For testing purposes, I’m now injecting the dependency so that callers don’t reference `GitShellHelper` directly but rather through the `ShellHelper` protocol.

To compensate for the fact that default arguments on the concrete implementation wouldn’t be seen in this setup, I added matching methods enclosing the default values in a `ShellHelper` protocol extension instead.

I hope the resulting code is clear, but I’m open to different approaches.

Reply:
I see. I guess the issue is that `Git.environment` is now a default of a generic `ShellHelper` which is odd IMO

Reply:
Thanks for your input, and apologies for the delay.

Fair point: the `ShellHelper` extension was `private` to the git implementation, but it may not have been obvious at the point of calling the methods with defaults.

As the goal is not to provide a generic `ShellHelper` abstraction, I moved one level down and abstracted `GitShellHelper`’s internal strategy instead (keeping the default arguments as-is). Hopefully, this new version should be clearer in the intentions (see the [updated diff](https://github.com/apple/swift-package-manager/pull/5941/files)).

If there are other objections to the general approach, I’m happy to take suggestions and go a different route.

I originally tried to implement an end-to-end test with no changes required to the code, but it looks like similar tests are currently disabled for reported flakiness (e.g. `testCanKillSubprocessOnSigInt`).

Alternatively, perhaps we could argue these changes may not require a test if unpractical to implement, but without being sure, I tend to err on the side of more tests :)

Let me know if you have a preferred approach. Thank you.